backend/frontend/views.py

```diff
@@ -113,16 +115,53 @@ def users_and_teams_nav(request):
     return render(request, "frontend/users_and_teams_nav/page.html", context)
 
 
-@user_passes_test(is_super)
+# @user_passes_test(is_super)
 def partial_teams_list(request):
+    user = request.user
+    checker = ObjectPermissionChecker(user)
+    print("hey", checker)
+    user_teams = user.teams()
     limit = 20
     current_team_count = int(request.GET.get("count", 0))
 
     all_teams = Team.objects.filter(active=True).order_by(
         "name"
     )  # TODO: only show teams that the current user is allowed to see
-    teams = all_teams[current_team_count : current_team_count + limit]
-    has_next_page = len(all_teams) > current_team_count + limit
+
+    is_superuser = user.is_superuser
+    is_assignee = any([team for team in all_teams if team in user_teams])
```

Comment:
This shouldn't be needed.  Users dont get to see the teams they are in unless they have permission to do so.

Reply:
Here is a better way to do it:

```
from guardian.shortcuts import get_objects_for_user

all_teams = Team.objects.filter(active=True).order_by(
        "name"
    ) 
if user.is_superuser:
    teams = all_teams
else:
    # we have to filter the teams
    teams = get_objects_for_user(
                user, Team.PERMISSION_VIEW, all_teams,
            )
```